File: BackEnd/Models/Endereco.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace MeAluga.Models
{
    [Owned]
    public class Endereco
    {   
        [MaxLength(10)]
        public string CEP { get; set; }

        [MaxLength(50)]
        public string Rua { get; set; }

        [Display(Name = "Número"), MaxLength(10)]
        public string Numero { get; set; }

        [MaxLength(50)]
        public string Bairro { get; set; }

        [MaxLength(100)]
        public string Complemento { get; set; }

        [MaxLength(50)]
        public string Cidade { get; set; }

        [MaxLength(50)]
        public string Estado { get; set; }
        
        public override string ToString()
        {
            return $"{Rua}, {Numero}, {Bairro}, {Complemento}, {CEP}, {Cidade} - {Estado}";
        }
    }
}

File: FrontEnd/src/domain/shared/Endereco.js
export default class Endereco{
    
    constructor(cep = '', rua = '', bairro = '', numero = '', complemento = '')
    {
        this.cep = cep;        
        this.rua = rua;
        this.bairro = bairro;
        this.numero = numero;
        this.complemento = complemento;
    }

}